pkg/client/interface.go

```diff
@@ -176,3 +178,23 @@ type TxClientOption func(TxClient)
 
 // SupplierClientOption defines a function type that modifies the SupplierClient.
 type SupplierClientOption func(SupplierClient)
+
+// QueryClientContext is used to distinguish a cosmosclient.Context intended
+// for use in queries from others.
+// This type is intentionally not an alias in order to make this distinction
+// clear to the dependency injector
+type QueryClientContext cosmosclient.Context
```

Comment:
Wdyt about renaming this to just  `QueryContext`? I feel like `Client` is redundant now that it's in the `client ` pkg.